config/routes.rb

```diff
@@ -3,6 +3,23 @@
   resources :companies do
     resources :jobs
   end
+
+  resources :categories, only: [:index, :show, :new, :create, :edit, :update, :destroy] do
+    resources :jobs
+  end
+
+  resources :jobs, only: [:show] do
+    resources :comments, only: [:new, :index, :create]
+  end
```

Comment:
Like that you understand that having jobs nested to categories doesn't mean you can't also have it out here.